### ISSUE TITLE
Freeze dropout mask for two evaluations?

### ISSUE THREAD

**Issue by RonMcKay** [keras-team/keras#9373]
Hi there,

I build a CNN (including two dropout layers) with the functional API. Then I computed a gradient (I use the sgd optimizer) of the error function for only one image with the corresponding label, with `learning_phase() = 1` to be in train mode. In contrary of what I have learned, every entry in the gradient is non zero. Shouldn't all weights that where dropped out also don't participate at the wheight update? If not, where is the part in the gradient calculation that changes from train to test mode?

Then I think there is no option to kind of "freeze" the seed of the dropout layers during evaluation? I want to make a forward pass with a single image, get the predicted class and then calculate the gradient of the cost function for the image with the predicted class, but with the same dropout mask as in the forward pass to get the predicted class.
